packages/frontend/app/services/channels/-utils/channel-saver.ts

```diff
@@ -0,0 +1,75 @@
+import Channel from 'emberclear/models/channel';
+import Vote from 'emberclear/models/vote';
+import VoteChain from 'emberclear/models/vote-chain';
+import ChannelContextChain from 'emberclear/models/channel-context-chain';
+import Identity from 'emberclear/models/identity';
+
+export async function saveChannel(channel: Channel) {
+  console.error('before saving channel admin');
+  await channel.admin.save();
+  console.error('after saving channel admin');
+
+  let membersPromises: Promise<Identity>[] = [];
+  channel.members.forEach((member) => membersPromises.push(member.save()));
```

Comment:
```ts
await Promise.all(channel.members.map((member) => member.save()))
```
would replace lines 12 through 14 ;)
(and for each of these promise arrays)

Reply:
Ah, yeah, that works better.

Reply:
Addressed in ae248b3